src/Internal.AspNetCore.Sdk/build/Git.targets

```diff
@@ -19,6 +21,29 @@
     </Sdk_GetGitCommitInfo>
   </Target>
 
+  <Target Name="CreateSourceLink"
+          Condition="'$(SourceLink)'=='' AND '$(GenerateSourceLinkFile)' != 'false' "
+          Inputs="$(IntermediateCommitHash);$(MSBuildAllProjects)"
+          Outputs="$(SourceLinkDestination)" >
+
+    <PropertyGroup>
+      <IsSourceProject>$(MSBuildProjectDirectory.StartsWith($(SourceDirectory)))</IsSourceProject>
```

Comment:
Can we remove this restriction, or does it cause problems? I know of at least one project that we ship that isn't in `src/`.

Reply:
This is my attempt to prevent us from needlessly sourcelinking test/sample/etc projects. Should I blacklist the `test` dir instead of whitelisting `src`, or do we not care about spending a little extra time generating these in places they won't get used?

Reply:
Ok. Easier to start small first. One change though: instead of using existence under `src/` use `IsPackable`. This property is define as `true` for any csproj that produces a nupkg.

Reply:
So, add this:
```xml
<GenerateSourceLinkFile Condition="'$(GenerateSourceLinkFile)' == '' AND '$(IsPackable)' != 'false'>true</GenerateSourceLinkFile>
```